core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala

```diff
@@ -137,8 +163,8 @@ protected[controller] class RestAPIVersion_v1(
             sendCorsHeaders {
                 (pathEndOrSingleSlash & get) {
                     complete(OK, info)
-                } ~ authenticate(basicauth) {
-                    user =>
+                } ~ authenticate(basicauth) { user =>
+                    checkPutSize(config, transid) {
```

Comment:
Did you consider moving this check into the inner routes? The disadvantage of doing it at this level is that you're checking every API request for every route even ones that do not support a PUT. Also, in the inner routes, there already is a method extraction and you can avoid doing that more than once.

In the future, I expect the entitlement check to play a role in these limits checks and that's another reason to move this into the inner routes. `AuthorizedRouteProvider.innerRoutes` in particular. More specifically, since this validation is applied to all PUTs of the `WhiskEntity` super type, then the idea place is in the `Entities` common API handler (which covers packages, triggers, rules and actions).

Here is an alternative implementation that you can compare:
https://github.com/openwhisk/openwhisk/commit/f602b90af00f8f51bf4372fd039916c198329baa
notes on the above link: I did not apply the other comments I made in this PR and I did not update the tests. If you think this is worthwhile, and you rebase your branch, I'll send you a pull request (I already rebased master and there are too many commits to cherry pick out).